include/RAJA/pattern/scan.hpp

```diff
@@ -31,202 +31,66 @@
-                    type_traits::is_iterator<Iter>,
-                    type_traits::is_iterator<IterOut>>
-exclusive_scan(const ExecPolicy &p,
-               Iter begin,
-               Iter end,
-               IterOut out,
-               Function binop = Function{},
-               T value = Function::identity())
+          typename Container,
+          typename Function = operators::plus<RAJA::detail::ContainerVal<Container>>,
+          typename Res = typename resources::get_resource<ExecPolicy>::type>
+RAJA_INLINE
+concepts::enable_if_t<resources::EventProxy<Res>,
+                      type_traits::is_execution_policy<ExecPolicy>,
+                      type_traits::is_range<Container>>
+inclusive_scan_inplace(ExecPolicy&& p,
+                       Container&& c,
+                       Function binop = Function{})
 {
-  using R = RAJA::detail::IterVal<IterOut>;
-  using U = RAJA::detail::IterVal<Iter>;
-  static_assert(type_traits::is_binary_function<Function, R, T, U>::value,
-                "Function must model BinaryFunction");
-  static_assert(type_traits::is_random_access_iterator<Iter>::value,
-                "Iterator must model RandomAccessIterator");
-  static_assert(type_traits::is_random_access_iterator<IterOut>::value,
-                "Output Iterator must model RandomAccessIterator");
-  if (begin == end) {
-    return;
-  }
-  impl::scan::exclusive(p, begin, end, out, binop, value);
+  auto r = Res::get_default();
+  return inclusive_scan_inplace(std::forward<ExecPolicy>(p),
+                                r,
+                                std::forward<Container>(c),
+                                binop);
 }
-
-// =============================================================================
-
-/*!
-******************************************************************************
-*
-* \brief  inclusive in-place scan execution pattern
-*
-* \param[in] p Execution policy
-* \param[in,out] c Random-Access Container
-* \param[in] binop binary function to apply for scan
-* \param[in] value identity value for binary function, binop
-*
-******************************************************************************
-*/
+///
 template <typename ExecPolicy,
+          typename Res,
           typename Container,
           typename Function = operators::plus<RAJA::detail::ContainerVal<Container>>>
-concepts::enable_if<type_traits::is_execution_policy<ExecPolicy>,
-                    type_traits::is_range<Container>>
-inclusive_scan_inplace(const ExecPolicy &p,
-                       Container &c,
+RAJA_INLINE
+concepts::enable_if_t<resources::EventProxy<Res>,
+                      type_traits::is_execution_policy<ExecPolicy>,
+                      type_traits::is_resource<Res>,
+                      type_traits::is_range<Container>>
+inclusive_scan_inplace(ExecPolicy&& p,
+                       Res& r,
+                       Container&& c,
                        Function binop = Function{})
```

Comment:
It seems that earlier nvccs are having trouble distinguishing these overloads.

Reply:
Just with inclusive_scan_implace or are you seeing this with other methods as well?

Reply:
It does affect the other sor and scan policy by value interface functions. Luckily @trws gave me a workaround that looks promising.